persistence/sql/persister_code.go

```diff
@@ -123,3 +111,39 @@ func useOneTimeCode[P any, U interface {
 
 	return target, nil
 }
+
+func incrementOTPCodeSubmitCount(ctx context.Context, p *Persister, flowID uuid.UUID, flowTableName string) (submitCount int, err error) {
+	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.incrementOTPCodeSubmitCount",
+		trace.WithAttributes(attribute.Stringer("flow_id", flowID), attribute.String("flow_table_name", flowTableName)))
+	defer otelx.End(span, &err)
+	defer func() {
+		span.SetAttributes(attribute.Int("submit_count", submitCount))
+	}()
+
+	nid := p.NetworkID(ctx)
+
+	if p.c.Dialect.Name() == "mysql" { // no RETURNING support
+		//#nosec G201 -- TableName is static
+		qUpdate := fmt.Sprintf("UPDATE %s SET submit_count = submit_count + 1 WHERE id = ? AND nid = ?", flowTableName)
+		//#nosec G201 -- TableName is static
+		qSelect := fmt.Sprintf("SELECT submit_count FROM %s WHERE id = ? AND nid = ?", flowTableName)
+		err = p.Transaction(ctx, func(ctx context.Context, tx *pop.Connection) error {
```

Comment:
Actually this is idiotic. There's no reason for this transaction to exist. MySQL should do a one-statement increment, and then a one-statement select, outside of a transaction.

Apologies...

Reply:
fixed